spootify/src/routes/Discover/components/Discover.js

```diff
@@ -1,27 +1,38 @@
 import React, { Component } from 'react';
 import DiscoverBlock from './DiscoverBlock/components/DiscoverBlock';
 import '../styles/_discover.scss';
+import { getCategories$, getFeaturedPlaylists$, getNewReleases$ } from '../../../common/utils/spotifyHttp';
 
-export default class Discover extends Component {
-  constructor() {
-    super();
-
-    this.state = {
-      newReleases: [],
-      playlists: [],
-      categories: []
-    };
-  }
-
-  render() {
-    const { newReleases, playlists, categories } = this.state;
-
-    return (
-      <div className="discover">
-        <DiscoverBlock text="RELEASED THIS WEEK" id="released" data={newReleases} />
-        <DiscoverBlock text="FEATURED PLAYLISTS" id="featured" data={playlists} />
-        <DiscoverBlock text="BROWSE" id="browse" data={categories} imagesKey="icons" />
-      </div>
-    );
-  }
+
+
+const Discover = () => {
+
+  const [newReleases, setNewReleases] = React.useState([]);
+  const [playlists, setPlaylists] = React.useState([]);
+  const [categories, setCategories] = React.useState([]);
+
+  React.useEffect(() => {
+    const sub$ = getNewReleases$().subscribe(setNewReleases);
+    return () => sub$.unsubscribe();
+  }, [])
+
+  React.useEffect(() => {
+    const sub$ = getFeaturedPlaylists$().subscribe(setPlaylists);
+    return () => sub$.unsubscribe();
+  }, [])
+
+  React.useEffect(() => {
+    const sub$ = getCategories$().subscribe(setCategories);
+    return () => sub$.unsubscribe();
+  }, [])
```

Comment:
Triggered three APIs in three `useEffect` so that they won't interfere with each other. And they will be triggered as quickly upon the React app's performance and browser's resource (might be in parallel or not).